File: ios/Packages/FeatureServices/PaymentService/PreparedPayment.swift
// Copyright (c). Gem Wallet. All rights reserved.

import Foundation
import Primitives

public struct PreparedPayment: Sendable {
    public let quotes: PaymentQuotes
    public let quote: PaymentQuote
    public let actions: [PaymentAction]
    public let isRelayed: Bool

    public init(quotes: PaymentQuotes, quote: PaymentQuote, actions: [PaymentAction], isRelayed: Bool) {
        self.quotes = quotes
        self.quote = quote
        self.actions = actions
        self.isRelayed = isRelayed
    }
}

File: android/gemcore/src/main/kotlin/com/gemwallet/android/model/PreparedPayment.kt
package com.gemwallet.android.model

import com.wallet.core.primitives.PaymentQuote
import com.wallet.core.primitives.PaymentQuotes
import uniffi.gemstone.PaymentAction

data class PreparedPayment(
    val quotes: PaymentQuotes,
    val quote: PaymentQuote,
    val actions: List<PaymentAction>,
    val isRelayed: Boolean,
)
